v2/api-validator/src/server/controllers/liquidity-controller.ts

```diff
@@ -0,0 +1,124 @@
+import { randomUUID } from 'crypto';
+import {
+  BadRequestError,
+  Quote,
+  QuoteCapability,
+  QuoteRequest,
+  QuoteStatus,
+  RequestPart,
+} from '../../client/generated';
+import { XComError } from '../../error';
+import { SUPPORTED_ASSETS, isKnownAsset } from './assets-controller';
+import _ from 'lodash';
+
+export class QuoteNotFoundError extends XComError {}
+export class InvalidQuoteRequestError extends XComError {
+  public requestPart = RequestPart.BODY;
+  constructor(
+    public message: string,
+    public errorType: BadRequestError.errorType,
+    public propertyName?: string
+  ) {
+    super(message);
+  }
+}
+
+const USERS_QUOTES_MAP: Map<string, Quote[]> = new Map();
+
+export const QUOTE_CAPABILITIES: QuoteCapability[] = [
+  { fromAsset: { assetId: SUPPORTED_ASSETS[0].id }, toAsset: { assetId: SUPPORTED_ASSETS[1].id } },
+  { fromAsset: { assetId: SUPPORTED_ASSETS[1].id }, toAsset: { assetId: SUPPORTED_ASSETS[0].id } },
+];
+
+function isSupportedLiquidityCapability(capability: QuoteCapability) {
+  return QUOTE_CAPABILITIES.some((quoteCapability) => _.isEqual(quoteCapability, capability));
+}
+
+export function validateQuoteRequest(quoteRequest: QuoteRequest): void {
+  if (!isKnownAsset(quoteRequest.fromAsset)) {
+    throw new InvalidQuoteRequestError(
+      'fromAsset is not a reference to a supported asset',
+      BadRequestError.errorType.SCHEMA_PROPERTY_ERROR,
+      'fromAsset'
+    );
+  }
+  if (!isKnownAsset(quoteRequest.toAsset)) {
+    throw new InvalidQuoteRequestError(
+      'toAsset is not a reference to a supported asset',
+      BadRequestError.errorType.SCHEMA_PROPERTY_ERROR,
+      'toAsset'
+    );
+  }
+  if (
+    !isSupportedLiquidityCapability({
+      fromAsset: quoteRequest.fromAsset,
+      toAsset: quoteRequest.toAsset,
+    })
+  ) {
+    throw new InvalidQuoteRequestError(
+      `${quoteRequest.fromAsset}/${quoteRequest.toAsset} conversion is not supported`,
+      BadRequestError.errorType.SCHEMA_ERROR
```

Comment:
This is not a `SCHEMA_ERROR`. It's, as you wrote yourself, "unsupported conversion".